prclz/reblock/reblock.py

```diff
@@ -92,12 +95,78 @@ def add_outside_node(block_geom: Polygon,
     
     return bldg_centroids 
 
-def reblock(parcels: MultiLineString,
+def snap_block(
+    block: Polygon,
+    parcels: gpd.GeoSeries,
+    ) -> LineString:
+    """
+    We need to identify which points in the parcels are from
+    the original block, which signifies existing streets. There
+    is a small eps difference in the points resulting from the
+    parcelization process so this maps each block point to 
+    its nearest vertex in the parcel. Note, we compare vertices
+    because downstream this will be used to identify points
+    in an iGraph graph, rather than a shapely geometry.
+
+    Args:
+        block: polygon representing existing streets
+        parcels: parcel boundaries
+
+    Returns:
+        A linestring repr of the block, where vertices 
+        have been mapped to those in parcels
+    """
+    if not isinstance(block, Polygon):
+        warning("block is type: %s", type(block))
+    block_bounds = block.boundary
+    parcel_bounds = parcels.boundary.unary_union
+
+    parcel_bounds_pts = []
+    for g in parcel_bounds:
+        parcel_bounds_pts.extend(list(g.coords))
+    parcel_bounds_pts = MultiPoint(parcel_bounds_pts)
+
+    if isinstance(block_bounds, LineString):
+        mapped_pts = [nearest_points(Point(b), parcel_bounds_pts)[1] for b in block_bounds.coords] 
+    else:
+        warning("block_bounds is type: %s", type(block_bounds))
+        mapped_pts = []
```

Comment:
stacked list comprehensions outperform `.append` and `.extend`, which is useful when we have a ton of points but leaving to you if you want to swap it

Reply:
Hmm that is interesting. Honestly there's probably other little things that maybe could be optimized but I sorta want to just leave it rather than make a little improvement and then break it. Leave it for the new RA's :)